deploy/kubernetes-1.13/pmem-csi-direct.yaml

```diff
@@ -233,6 +233,7 @@ spec:
                   "-deviceManager=ndctl", 
                   "-endpoint=$(CSI_ENDPOINT)",
                   "-nodeid=$(KUBE_NODE_NAME)",
+                  "-statePath=/csi/volumes",
```

Comment:
This isn't a suitable location. The `plugin-socket-dir` should be used for interaction with kubelet and sidecars, not as private data store for the driver. The data then ends up in /var/lib/kubelet, which is a directory that might get wiped when reinstalling the cluster.

As PMEM survives a cluster reinstallation (unless wiped, too, see blow), the persistent data also must survive.

I suggest to use `/var/lib/pmem-csi.intel.com` as state directory.

Cluster administrators will need to know about this. Please add documentation. It needs to explain that a) we have persistent state outside of the PMEM itself b) where that is c) how to reset the node (re-initialize PMEM, clean the state directory.

Reply:
> I suggest to use `/var/lib/pmem-csi.intel.com` as state directory.

Who is supposed to create this directory on node host? Do we expect that the admin would create this on every node before deploying the driver, or it must be created by the driver itself?. If its driver, then our daemonset has to mount '/var/lib' to the driver container.

Reply:
I would say the driver. It already runs as a privileged pod, so whether it also has access to `/var/lib` won't make a big difference.

Reply:
Actually, we can limit access to just the dir inside `/var/lib` with:
```
      volumes:
        - hostPath:
            path: /var/lib/pmem-csi.intel.com
            type: DirectoryOrCreate
          name: pmem-state-dir
```

Reply:
> e: DirectoryOrCreate
>           name: pmem-state-dir

Thanks, I will give it a try.

Reply:
Done

Reply:
> Cluster administrators will need to know about this. Please add documentation. It needs to explain that a) we have persistent state outside of the PMEM itself b) where that is c) how to reset the node (re-initialize PMEM, clean the state directory.

@pohly  What is the right place to document it. Shall we take this part of "cluster admin document", which is currently missing.

Reply:
I would start a new section in the README.md for it. When it gets too large, we can still split it out later.